### PR TITLE
Report CV scores from within OptunaSearchCV

### PR DESCRIPTION
**Pull request by adjeiv**
## Motivation
Refs #4958 

## Description of the changes
Call report_cross_validation_scores from within OptunaSearchCV, so that if we pass in a terminator callback (e.g. using CrossValidationErrorEvaluator), we can use the scores to determine whether to terminate the study.

Tested with the following adapted sample code:

```
import optuna
from sklearn.datasets import load_iris
from sklearn.svm import SVC
from optuna.terminator import TerminatorCallback, Terminator, CrossValidationErrorEvaluator

clf = SVC(gamma="auto")
param_distributions = {
    "C": optuna.distributions.FloatDistribution(1e-10, 1e10, log=True)
}
terminator = TerminatorCallback(terminator=Terminator(error_evaluator=CrossValidationErrorEvaluator(), min_n_trials=4))
optuna_search = optuna.integration.OptunaSearchCV(clf, param_distributions, callbacks=[terminator])
X, y = load_iris(return_X_y=True)
optuna_search.fit(X, y)
```

### CODE REVIEW

**Comment by HideakiImamura** [comment]
@Alnusjaponica @nabenabe0928 Could you review this PR?

**Review by nabenabe0928 — APPROVED** [review]
Thank you for the PR, LGTM!

File: optuna/integration/sklearn.py [hunk 1]
```diff
@@ -243,6 +244,10 @@ def __call__(self, trial: Trial) -> float:
 
         self._store_scores(trial, scores)
 
+        test_scores = scores["test_score"]
```
Comment (Alnusjaponica) [inline optuna/integration/sklearn.py:247]: The `cross_validate()` function does not necessarily return a dict with "test_score" key, but it is only when multiple metrics are given though `scoring` parameter, which does not happen in this case.

**Review by Alnusjaponica — APPROVED** [review]
LGTM. We really appreciate your contribution.

**Comment by Alnusjaponica** [comment]
Let me merge this PR after I confirm that `windows-tests` passes.

**Pull request merged**